parse.go

```diff
@@ -139,18 +144,21 @@ func (p *Parser) GetMetadata() Dataset {
 
 // readHeader reads the DICOM magic header and group two metadata elements.
 func (p *Parser) readHeader() ([]*Element, error) {
-	// Must read as LittleEndian explicit VR
-	err := p.reader.Skip(128) // skip preamble
-	if err != nil {
-		log.Println("skip er")
-		return nil, err
+	// Check to see if magic word is at byte offset 128. If not, this is a
+	// non-standard non-compliant DICOM. We try to read this DICOM in a
+	// compatibility mode, where we rewind to position 0 and blindly attempt to
+	// parse a Dataset (and do not parse metadata in the usual way).
+	data, err := p.reader.Peek(128 + 4) // the last 4 bytes should be "DICM"
```

Comment:
Let's maybe check `err` here. Looking at Golang docs, it's possible this thing can return an error.

Reply:
yes it can return an error, I was planning on cleaning this up (among some other items) before merging this, but thanks for flagging! Got distracted with some other PRs.